logger/log.go

```diff
@@ -0,0 +1,73 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package logger
```

Comment:
Do not put `logger` in the root directory  :)
It is recommended to adjust the directory structure, such as `/pkg/logger`.

Reply:
Where is this recommendation from? Could you share the link? I did a quick search, it says otherwise, https://stackoverflow.com/questions/47369621/what-is-the-use-of-pkg-directory-in-go

Reply:
> Library code that's ok to use by external applications

From here, https://github.com/golang-standards/project-layout/blob/master/pkg/README.md.

Can't see we have things to provide external.

Reply:
https://github.com/golang-standards/project-layout#go-directories

I agree to adjust the directory structure, but am not sure whether or not we should put the `logger` into `/pkg`, as 

> `/pkg`: Library code that's ok to use by external applications

is the logger expected to be used by others?

Reply:
Maybe we can move `logger` into `/internal`?

Reply:
I've done some refactoring to support  others using, but I prefer to put it on /internal